tinygrad/codegen/linearizer.py

```diff
@@ -224,9 +224,29 @@ def linearize(self):
 
     # define indexes
     global_idxs, loop_global_idxs = get_grouped_dims("gidx", 0, self.full_shape[:self.global_dims], 3 if self.opts.has_local else 0)
-    local_idxs, loop_local_idxs = get_grouped_dims("lidx", self.global_dims, self.full_shape[self.global_dims:self.first_reduce+len(self.group_for_reduce)], 3 if self.opts.has_local else 0)
-    full_upcast_idxs = [Variable(None, 0, s-1) for s in self.full_shape[self.shape_len-self.upcasted:]]
-    upcast_idxs = [Variable(None, 0, s-1) for s in self.output_shape[self.shape_len-self.upcasted:]]
+    local_idxs, loop_local_idxs = get_grouped_dims("lidx", self.global_dims+(1 if self.tensor_core else 0), self.full_shape[self.global_dims:self.first_reduce+len(self.group_for_reduce)], (2 if self.tensor_core else 3) if self.opts.has_local else 0)
+    full_upcast_idxs = [Variable(None, 0, s-1) for s in self.full_shape[self.shape_len-self.upcasted+(3 if self.tensor_core else 0):]]
+    upcast_idxs = [Variable(None, 0, s-1) for s in self.output_shape[self.shape_len-self.upcasted+(3 if self.tensor_core else 0):]]
+    warp_idxs, aliased_wmma_idxs = [], []
+
+    if self.tensor_core:
+      warp_idxs, loop_warp_idxs = get_grouped_dims("lidx", self.global_dims, [prod(self.tensor_core.threads)], 1)
+      warp_segment_idxs = []
+      for segment_sz in self.tensor_core.threads:
+        warp_segment_idxs.append(loop_warp_idxs[0] % segment_sz)
+        loop_warp_idxs[0] //= segment_sz
+      def calc_tc_idxs(local_size: int, aliases: List[List[int]]):
+        idxs = []
+        for alias in aliases:
+          full_var, full_var_sz = Variable.num(0), 1
+          if alias[0] != 0:
+            for i in alias:
+              next_var = warp_segment_idxs[i-1] if i > 0 else Variable(None, 0, local_size-1)
+              full_var += next_var * full_var_sz
+              full_var_sz *= next_var.max+1
+          idxs.append(full_var)
+        return idxs
+      aliased_wmma_idxs = [calc_tc_idxs(x[0], x[1]) for x in zip(self.tensor_core.thread_local_sizes, self.tensor_core.thread_local_aliases)]
```

Comment:
`[calc_tc_idxs(s, a) for s, a in zip(self.tensor_core.thread_local_sizes, self.tensor_core.thread_local_aliases)]`
or
``[calc_tc_idxs(*x) for x in zip(self.tensor_core.thread_local_sizes, self.tensor_core.thread_local_aliases)]`

Reply:
Done!